core/src/test/java/org/apache/iceberg/TestTables.java

```diff
@@ -190,9 +191,34 @@ public void commit(TableMetadata base, TableMetadata updatedMetadata) {
             throw new CommitFailedException("Injected failure");
           }
           Integer version = VERSIONS.get(tableName);
+          // remove changes from the committed metadata
+          if (updatedMetadata.changes().isEmpty()) {
+            this.current = updatedMetadata;
+          } else {
+            this.current = new TableMetadata(
```

Comment:
Since this `TableMetadata` doesn't get written to JSON to reset changes, this resets them manually instead.